config/config.go

```diff
@@ -41,6 +43,38 @@ func Config() *ConfigStruct {
 	return configData
 }
 
+//setMapValueEnvOverride recursively iterate over the keys and updates the map value accordingly
+func setMapValueEnvOverride(keys []string, mapValue map[string]interface{}, value interface{}) map[string]interface{} {
```

Comment:
returning a map[string]interface{} implies you are returning a copy of the original, and the original is unmodified. I suggest removing this return value and accepting that the map will be modified in place.

Reply:
Done